exporter/fileexporter/file_exporter.go

```diff
@@ -50,33 +59,44 @@ func newFileExporter(conf *Config) *fileExporter {
 			MaxBackups: conf.Rotation.MaxBackups,
 			LocalTime:  conf.Rotation.LocalTime,
 		},
+		isJSON:           isJSONData(conf),
+		tracesMarshaler:  tracesMarshaler,
+		metricsMarshaler: metricsMarshaler,
+		logsMarshaler:    logsMarshaler,
 	}
 }
 func (e *fileExporter) Capabilities() consumer.Capabilities {
 	return consumer.Capabilities{MutatesData: false}
 }
 
 func (e *fileExporter) ConsumeTraces(_ context.Context, td ptrace.Traces) error {
-	buf, err := tracesMarshaler.MarshalTraces(td)
+	buf, err := e.tracesMarshaler.MarshalTraces(td)
 	if err != nil {
 		return err
 	}
-	return exportMessageAsLine(e, buf)
+	return exportMessage(e, buf)
 }
 
 func (e *fileExporter) ConsumeMetrics(_ context.Context, md pmetric.Metrics) error {
-	buf, err := metricsMarshaler.MarshalMetrics(md)
+	buf, err := e.metricsMarshaler.MarshalMetrics(md)
 	if err != nil {
 		return err
 	}
-	return exportMessageAsLine(e, buf)
+	return exportMessage(e, buf)
 }
 
 func (e *fileExporter) ConsumeLogs(_ context.Context, ld plog.Logs) error {
-	buf, err := logsMarshaler.MarshalLogs(ld)
+	buf, err := e.logsMarshaler.MarshalLogs(ld)
 	if err != nil {
 		return err
 	}
+	return exportMessage(e, buf)
+}
+
+func exportMessage(e *fileExporter, buf []byte) error {
+	if !e.isJSON {
```

Comment:
Rather than check the format type, we should use save the function we want to call on the exporter. This way it only needs to be checked once.